stdlibs/fetch.py

```diff
@@ -353,9 +359,58 @@ def try_parse(path: Path, data: Optional[bytes] = None) -> cst.Module:
     raise parse_error or Exception("unknown parse failure")
 
 
+def regen_urls() -> None:
+    base_url = "https://www.python.org/ftp/python/"
+    body = urlopen(base_url).read()
+    max_version_dirs: Dict[str, Version] = {}
+    for subdir in OVERLY_SIMPLISTIC_LINK_RE.findall(body.decode()):
+        if m := DOTTED_VERSION_DIR.fullmatch(subdir):
+            v = m.group(1) + "." + m.group(2)
+            vo = Version(subdir[:-1])
+            if vo < Version("2.3"):
+                continue
+
+            if v not in max_version_dirs or vo > max_version_dirs[v]:
+                max_version_dirs[v] = vo
+
+    new_urls = []
+    for x, y in max_version_dirs.items():
+        body = urlopen(base_url + str(y) + "/").read()
+        archives = OVERLY_SIMPLISTIC_LINK_RE.findall(body.decode())
+        tars = []
+        for file in archives:
+            if m := PYTHON_ARCHIVE_RE.fullmatch(file):
+                tars.append((Version(m.group(1)), str(y) + "/" + file))
+        new_urls.append((Version(x), max(tars)[1]))
+
+    new_urls.sort()
+
+    block = ""
+    for v, url in new_urls:
+        block += f'    "{str(v)}": "{base_url}{url}",\n'
+
+    buf = []
+    state = 0
+    for line in open(__file__):
+        if state == 0 and line.strip() == "RELEASES = {":
+            state = 1
+            buf.append(line)
+            buf.extend(block)
+        elif state == 1 and line.strip() == "}":
+            state = 2
+            buf.append(line)
+        elif state != 1:
+            buf.append(line)
+
+    assert state == 2, state
+    open(__file__, "w").write("".join(buf))
```

Comment:
Maybe it's time to move those URLs into a separate JSON or TOML formatted file so that we can simplify this code? Happy to pick up that part if you'd prefer.

Reply:
Done.  I don't think the data needs a copyright notice.

I also intentionally didn't update the year in the fetch script, because then all the generated files change.  Feel free to when you release though, especially if you want to wait until January.